src/parser/symtab.ts

```diff
@@ -0,0 +1,1738 @@
+        this.visitExpr(outermost.iter);
+
+        /* Create comprehension scope for the rest */
+        this.enterBlock(scopeName, FunctionBlock, e, e.lineno, e.col_offset, e.end_lineno, e.end_col_offset);
+
+        if (outermost.is_async) {
+            this.cur.coroutine = true;
+        }
+
+        this.cur.comprehension = true;
+
+        /* Outermost iter is received as an argument */
+        this.implicitArg(0);
+
+        /* Visit iteration variable target, and mark them as such */
+        this.cur.comp_iter_target = true;
+        this.visitExpr(outermost.target);
+        this.cur.comp_iter_target = false;
+
+        /* Visit the rest of the comprehension body */
+        this.SEQ(this.visitExpr, outermost.ifs);
+        this.SEQTail(this.visitComprehension, generators, 1);
+        if (value) {
+            this.visitExpr(value);
+        }
+
+        this.visitExpr(elt);
+
+        if (this.cur.generator) {
+            throw new pySyntaxError(
+                e._kind === ASTKind.ListComp
+                    ? "'yield' inside list comprehension"
+                    : e._kind === ASTKind.SetComp
+                    ? "'yield' inside set comprehension"
+                    : e._kind === ASTKind.DictComp
+                    ? "'yield' inside dict comprehension"
+                    : "'yield' inside generator expression",
+                [this.filename, e.lineno, e.col_offset, ""]
+            );
+        }
+
+        this.cur.generator = isGenerator;
+        const isAsync = this.cur.coroutine && !isGenerator;
+        this.exitBlock();
+        if (isAsync) {
+            this.cur.coroutine = true;
+        }
+    }
+
+    visitKeyword(k: keyword) {
+        this.visitExpr(k.value);
+    }
+
+    visitGenexp(e: GeneratorExp) {
+        return this.handleComprehension(e, "genexpr", e.generators, e.elt, null);
+    }
+
+    visitListcomp(e: ListComp) {
+        return this.handleComprehension(e, "listcomp", e.generators, e.elt, null);
+    }
+
+    visitSetcomp(e: SetComp) {
+        return this.handleComprehension(e, "setcomp", e.generators, e.elt, null);
+    }
+
+    visitDictcomp(e: DictComp) {
+        return this.handleComprehension(e, "dictcomp", e.generators, e.key, e.value);
+    }
+
+    visitComprehension(lc: comprehension) {
+        assert(this.cur);
+        this.cur.comp_iter_target = true;
+        this.visitExpr(lc.target);
+        this.cur.comp_iter_target = false;
+        this.cur.comp_iter_expr++;
+        this.visitExpr(lc.iter);
+        this.cur.comp_iter_expr--;
+        this.SEQ(this.visitExpr, lc.ifs);
+        if (lc.is_async) {
+            this.cur.coroutine = true;
+        }
+    }
+
+    visitExpr(e: expr) {
+        // @stu todo maybe?
+        // if (++st->recursion_depth > st->recursion_limit) {
+        //     PyErr_SetString(PyExc_RecursionError,
+        //                     "maximum recursion depth exceeded during compilation");
+        //     VISIT_QUIT(st, 0);
+        // }
```

Comment:
We might just throw our own recursion error from javascript which will throw a recursion error in python anyway.

Reply:
so I can remove all this recursion stuff?

Reply:
I think so.